appyx-components/modal/common/src/commonMain/kotlin/com/bumble/appyx/components/modal/operation/Revert.kt

```diff
@@ -0,0 +1,38 @@
+package com.bumble.appyx.components.modal.operation
+
+import androidx.compose.animation.core.AnimationSpec
+import com.bumble.appyx.components.modal.Modal
+import com.bumble.appyx.components.modal.ModalModel
+import com.bumble.appyx.interactions.Parcelize
+import com.bumble.appyx.interactions.core.model.transition.BaseOperation
+import com.bumble.appyx.interactions.core.model.transition.Operation
+
+@Parcelize
+class Revert<InteractionTarget : Any>(
+    override var mode: Operation.Mode = Operation.Mode.KEYFRAME
+) : BaseOperation<ModalModel.State<InteractionTarget>>() {
+
+    override fun isApplicable(state: ModalModel.State<InteractionTarget>) = true
+
+    override fun createTargetState(fromState: ModalModel.State<InteractionTarget>) =
+        if (fromState.fullScreen != null || fromState.modal != null) {
```

Comment:
If you have a condition, it should always be expressed in the `isApplicable` method too.

Reply:
Refactored.